action.yaml

```
@@ -0,0 +1,51 @@
+name: Elixir Build and Cache Dependencies
+inputs:
+  elixir-version:
+    required: true
+    type: string
+  erlang-version:
+    required: true
+    type: string
+  mix-env:
+    required: true
+    type: string
+outputs:
+  cache-key:
+    value: ${{ steps.compute-cache-key.outputs.cache-key }}
+  cache-key-suffix:
+    value: ${{ steps.compute-cache-key.outputs.cache-key-suffix }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```

`fetch-depth: 0` gets the whole history from the repository. I suspect all you need is whatever ref is current when this action is called (usually `HEAD` of some branch). If that's the case, you can remove `fetch-depth` and `with:`.
```suggestion
    - uses: actions/checkout@v2
```

---

Updated as suggested.